### DISCUSSION TITLE
inference on Single Image

### DISCUSSION THREAD

**Discussion by mohandeeprdori** [open-edge-platform/anomalib#2268] [Q&A] [answered]
I have a fastflow model with a ckpt file. i want to load this model and run inference on images. The problem is im able to do it only in batches and get the results of all images together. Instead, i want to have results of one image at a time. Can anyone help me with this?
@ashwinvaidya17

**Comment by ashwinvaidya17** [comment, marked as answer]
Do you have some sample code of how you are approaching this? Does `anomalib predict` not give you the desired result?

**Reply by mohandeeprdori** [reply]
I was trying to write a piece of code instead of doing it in the cli. I got the solution by calling the PredictDataset class and creating a dataloader out of a single image and do engine.predict to get the predictions score, masks etc.But when we give a datamodule with a specific category, predictions and results are returned in batch.
